sql/api/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3911,6 +3911,44 @@ object functions {
   def encode(value: Column, charset: String): Column =
     Column.fn("encode", value, lit(charset))
 
+  /**
+   * Returns true if the input is a valid UTF-8 string, otherwise returns false.
+   *
+   * @group string_funcs
+   * @since 4.0.0
+   */
+  def is_valid_utf8(str: Column): Column =
+    Column.fn("is_valid_utf8", str)
+
+  /**
+   * Returns a new string in which all invalid UTF-8 byte sequences, if any, are replaced by the
+   * Unicode replacement character (U+FFFD).
+   *
+   * @group string_funcs
+   * @since 4.0.0
+   */
+  def make_valid_utf8(str: Column): Column =
+    Column.fn("make_valid_utf8", str)
+
+  /**
+   * Returns the input value if it corresponds to a valid UTF-8 string, or emits an error
```

Comment:
which error does it emit?

Reply:
`INVALID_UTF8_STRING` (`invalidUTF8StringError`)

Reply:
Let's reflect that in the comment.

Reply:
note that it's usually not reflected in the comments, although I see 3 instances where the exception is specified